packages/amplify-graphql-generation-transformer/src/resolvers/invoke-bedrock-resolver-fn.template.js

```diff
@@ -28,18 +30,31 @@ export function response(ctx) {
   if (ctx.error) {
     util.error(ctx.error.message, ctx.error.type);
   }
-  const body = JSON.parse(ctx.result.body);
-  const { content } = body.output.message;
 
-  if (content.length < 1) {
-    util.error('No content block in assistant response.', 'error');
+  // Check for AccessDeniedException.
+  // This can happen if:
+  //   1. The Bedrock model isn't enabled in this region.
+  //   2. The IAM policy statement for the role assumed by the data source for this resolver doesn't include them model.
+  //    This shouldn't happen because we're managing the policy statements.
+  // We're using a generic error description here (as opposed to using the response body)
+  // to prevent information about the system from leaking.
+  const errorType = ctx.result.headers['x-amzn-ErrorType'];
+  if (errorType) {
+    if (errorType.startsWith('AccessDeniedException')) {
+      const errorMessage = 'The model is disabled or this generation route is missing a necessary identity-based policy.';
+      util.error(errorMessage, 'AccessDeniedException');
```

Comment:
Not your problem, but I have to consult the docs every time to remember that `util.error` actually throws rather than just invoking a function and continuing to process. `</rant>`

Reply:
yea, same here.